cli/src/main.rs

```diff
@@ -256,7 +256,10 @@ fn override_configuration(mut config: Config, options: &Args) -> anyhow::Result<
         if let Some(go_package) = options.go_package.as_ref() {
             config.go.package = go_package.to_string();
         }
-        assert_go_package_present(&config)?;
+
+        if let Some(args::AvailableLanguage::Go) = options.language {
+            assert_go_package_present(&config)?;
+        }
```

Comment:
The whole block for `#[cfg(feature = "go")]` can be condensed to:

```rust
    #[cfg(feature = "go")]
    {
        use std::ops::Not;
        if let Some(go_package) = options.go_package.as_ref() {
            config.go.package = go_package.to_string();
        }

        if matches!(options.language, Some(args::AvailableLanguage::Go)) {
            anyhow::ensure!(
                config.go.package.is_empty().not(),
               "Please provide a package name in the typeshare.toml or using --go-package <package name>"
            );
        }
    }
```

The `assert_go_package_present` can then be removed.

Reply:
Is there a reason why we specifically used the `.not()` method and not just do
```
!config.go.package.is_empty()
```
From my understanding, they are the same

Reply:
Yup I removed that in my edit.

Reply:
Fixed!